testnet/register_node.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+# NODE_MASK is set via compose.yml file.
+
+node_key_from_id() {
+	LENGTH=$(echo -n $NODE_ID | wc -c)
+	echo $(echo $NODE_MASK | sed "s/.\{${LENGTH}\}$/${NODE_ID}/")
+}
+
+END=$DOCKER_REPLICAS
+NODE_ID=1
+NODE_IP=$(hostname -i)
+NODE_KEY=$(node_key_from_id)
+
+register_node() {
+	## Conditional transaction to set node config key if it doesn't exist.
+	## Newlines in EOF block are important, more info here:
+	## https://github.com/etcd-io/etcd/tree/main/etcdctl#examples-3
+	etcdctl txn <<EOF
+mod("/node/${NODE_ID}") = "0"
+
+put /node/${NODE_ID} "${NODE_ID}"
+put /config/node/${NODE_ID}/key "${NODE_KEY}"
+put /config/node/${NODE_ID}/ip "${NODE_IP}"
+
+
+EOF
+}
+
+while [ $NODE_ID -le $END ]; do
```

Comment:
You need to quote variable usage, and use curly braces.
```suggestion
while [ "${NODE_ID}" -le "${END}" ]; do
```
I would recommend running [`shellcheck`](https://www.shellcheck.net/) against your scripts.